includes/Core/Authentication/Verification_Meta.php

```diff
@@ -20,105 +19,40 @@
 	 * User option key.
 	 */
 	const OPTION = 'googlesitekit_site_verification_meta';
 
 	/**
-	 * User_Options object.
+	 * Sets the value of the setting with the given value.
 	 *
-	 * @since 1.0.0
-	 * @var User_Options
-	 */
-	private $user_options;
-
-	/**
-	 * Transients object.
-	 *
-	 * @since 1.0.0
-	 * @var Transients
-	 */
-	private $transients;
-
-	/**
-	 * Constructor.
+	 * @since n.e.x.t
 	 *
-	 * @since 1.0.0
+	 * @param mixed $value Setting value. Must be serializable if non-scalar.
 	 *
-	 * @param User_Options $user_options User Options instance.
-	 * @param Transients   $transients   Transients instance.
-	 */
-	public function __construct( User_Options $user_options, Transients $transients ) {
-		$this->user_options = $user_options;
-		$this->transients   = $transients;
-	}
-
-	/**
-	 * Retrieves the user verification tag.
-	 *
-	 * @since 1.0.0
-	 *
-	 * @return string|bool Verification tag, or false if not set.
-	 */
-	public function get() {
-		return $this->user_options->get( self::OPTION );
-	}
-
-	/**
-	 * Saves the user verification tag.
-	 *
-	 * @since 1.0.0
-	 *
-	 * @param string $meta_tag Meta tag to store.
 	 * @return bool True on success, false on failure.
 	 */
-	public function set( $meta_tag ) {
-		$status = $this->user_options->set( self::OPTION, $meta_tag );
-		if ( $status ) {
-			$this->transients->delete( 'googlesitekit_verification_meta_tags' );
-		}
-		return $status;
-	}
+	public function set( $value ) {
+		/**
+		 * Triggers the meta tag cache to be cleared.
+		 */
+		do_action( 'googlesitekit_invalidate_verification_meta_cache' );
 
-	/**
-	 * Checks whether a verification tag for the user is present.
-	 *
-	 * @since 1.0.0
-	 *
-	 * @return bool True if verification meta tag is set, false otherwise.
-	 */
-	public function has() {
-		$meta_tag = (string) $this->get();
-		return ! empty( $meta_tag );
+		return parent::set( $value );
 	}
 
+
 	/**
-	 * Gets all available verification tags for all users.
-	 *
-	 * This is a special method needed for printing all meta tags in the frontend.
+	 * Gets the underlying meta key for the verification meta.
 	 *
-	 * @since 1.0.0
+	 * @since n.e.x.t
 	 *
-	 * @return array List of verification meta tags.
+	 * @return string
 	 */
-	public function get_all() {
-		global $wpdb;
-
-		$meta_tags = $this->transients->get( 'googlesitekit_verification_meta_tags' );
-
-		if ( false === $meta_tags ) {
-			$meta_key = self::OPTION;
-			if ( ! \Google\Site_Kit\Plugin::instance()->context()->is_network_mode() ) {
-				$meta_key = $wpdb->get_blog_prefix() . $meta_key;
-			}
-			// phpcs:ignore WordPress.DB.DirectDatabaseQuery
-			$meta_tags = $wpdb->get_col( $wpdb->prepare( "SELECT DISTINCT meta_value FROM {$wpdb->usermeta} WHERE meta_key = %s", $meta_key ) );
-			$this->transients->set( 'googlesitekit_verification_meta_tags', $meta_tags );
-		}
-
-		return (array) $meta_tags;
+	public function get_meta_key() {
+		return $this->user_options->get_meta_key( self::OPTION );
```

Comment:
This method should not be here. The `Site_Verification` class should rather just use `User_Options::get_meta_key( $option )` for that.